examples/rpc.py

```diff
@@ -0,0 +1,219 @@
+from livekit import rtc, api
+import os
+import json
+import asyncio
+from dotenv import load_dotenv
+
+load_dotenv(dotenv_path=".env.local", override=False)
+LIVEKIT_API_KEY = os.getenv("LIVEKIT_API_KEY")
+LIVEKIT_API_SECRET = os.getenv("LIVEKIT_API_SECRET")
+LIVEKIT_URL = os.getenv("LIVEKIT_URL")
+if not LIVEKIT_API_KEY or not LIVEKIT_API_SECRET or not LIVEKIT_URL:
+    raise ValueError(
+        "Missing required environment variables. Please check your .env.local file."
+    )
+
+
+async def main():
+    room_name = f"rpc-test-{os.urandom(4).hex()}"
+
+    print(f"Connecting participants to room: {room_name}")
+
+    callers_room, greeters_room, math_genius_room = await asyncio.gather(
+        connect_participant("caller", room_name),
+        connect_participant("greeter", room_name),
+        connect_participant("math-genius", room_name),
+    )
+
+    register_receiver_methods(greeters_room, math_genius_room)
+
+    try:
+        print("\n\nRunning greeting example...")
+        await asyncio.gather(perform_greeting(callers_room))
+    except Exception as error:
+        print("Error:", error)
+
+    try:
+        print("\n\nRunning error handling example...")
+        await perform_divide(callers_room)
+    except Exception as error:
+        print("Error:", error)
+
+    try:
+        print("\n\nRunning math example...")
+        await perform_square_root(callers_room)
+        await asyncio.sleep(2)
+        await perform_quantum_hypergeometric_series(callers_room)
+    except Exception as error:
+        print("Error:", error)
+
+    print("\n\nParticipants done, disconnecting...")
+    await callers_room.disconnect()
+    await greeters_room.disconnect()
+    await math_genius_room.disconnect()
+
+    print("Participants disconnected. Example completed.")
+
+
+def register_receiver_methods(greeters_room: rtc.Room, math_genius_room: rtc.Room):
+    @greeters_room.local_participant.rpc_method("arrival")
+    async def arrival_method(
+        request_id: str,
+        caller_identity: str,
+        payload: str,
+        response_timeout_ms: int,
```

Comment:
I think we should put that inside a dataclass like RpcData.
Otherwise it isn't future proof and may break if we want to add a new field in the future

Reply:
Ppl may also ignore some args, and Python will raise an exception when trying to call a function with more arguments than what the function defines

Reply:
I'm not really sure a dataclass is better. it adds more indirection to the documentation and increases the surface area of imports needed. 

Why not just make these kwargs only? I changed it so you must call this method with kwargs (by adding a `*` before them, which works in python 3). If we introduce new ones in the future that's fine, we can ensure they have a default value which is the same as with a data class.

What do you think?

Reply:
What I mean is that if if the user defines its callback this way:

```
def my_callback(arg1, arg2):
   pass

```

and in our code we call the callback with 3 args `my_callback(arg1, arg2, arg3)`. Python will raise an exception. So the code isn't future-proof since as soon as we add a new argument, it'll break

Reply:
oh you mean on the handler - ok will take a look

Reply:
this was a good call - made that change and will go make it everywhere else as well